core/state/statedb.go

```diff
@@ -576,7 +576,7 @@ func (s *StateDB) updateStateObject(obj *stateObject) {
 	}
 	// Encode the account and update the account trie
 	addr := obj.Address()
-	if err := s.trie.UpdateAccount(addr, &obj.data, len(obj.code)); err != nil {
+	if err := s.trie.UpdateAccount(addr, &obj.data, len(s.GetCode(addr))); err != nil {
```

Comment:
The problem appeared in the dumped state for the 2935 contract.
In the post-state, the codeLen encoded in basicData was 0 which made no sense.

After looking, the problem is that `obj.code` was nil since technically the EVM never called `GetCode(...)`. `GetCode(...)` actually fills `obj.code` as a cache.

Since we were accessing the cache directly, we were re-encoding basicdata with length 0 which was wrong.

Reply:
Normal contract would have already call `GetCode(...)` in the EVM, so this extra call will simply return it from the cache.

But this will cover for system contracts (like 2935) which update from storage directly as an optimization.